### ISSUE TITLE
frame-step plays audio

### ISSUE THREAD

**Issue by Qwerty-Space** [mpv-player/mpv#6104]
### mpv version and platform
Windows, not sure what version:
![image](https://user-images.githubusercontent.com/6967592/44961492-e8cb3c80-af09-11e8-8aa6-c2b1ad04ac94.png)

### Reproduction steps
Hold down `frame-step` key

### Expected behavior
Cycle through frames

### Actual behavior
It pretty much just plays the video, audio and all.  Sound should be muted when using this button

### Log file
https://b.va-online.moe/uzapuqusem.hs

**Comment by garoto** [comment]
[**@Qwerty-Space**](https://github.com/Qwerty-Space) commented on [Sep 2, 2018, 7:49 PM GMT-3](https://github.com/mpv-player/mpv/issues/6104#issue-356320222 "2018-09-02T22:49:00Z - Replied by Github Reply Comments"):
> ### mpv version and platform
> 
> Windows, not sure what version:  
> 
> [![image](https://user-images.githubusercontent.com/6967592/44961492-e8cb3c80-af09-11e8-8aa6-c2b1ad04ac94.png)](https://user-images.githubusercontent.com/6967592/44961492-e8cb3c80-af09-11e8-8aa6-c2b1ad04ac94.png)
> 
Invoking `mpv.exe` directly does not spawn its CLI interface. Try a simple `mpv` or, on a worst case scenario, `mpv.com`.

Regarding the audio playing while frame-stepping, @wm4 (RIP) had this to say about a similar issue regarding frame-stepping: https://github.com/mpv-player/mpv/issues/3476#issuecomment-244064404

**Comment by haasn** [comment]
FWIW I always thought this was a feature, not a bug?

**Comment by garoto** [comment]
I always wished for `input-ar-rate` to be respected while doing `frame-step` (audio or not was never relevant for me). Sort of a silly request on the surface I imagine, but nowadays I somewhat can see how doing the work to implement any these sort of user-requests can bite you in the ass in the future.

**Comment by avih** [comment]
> Thanks for the input. Would be nice if it muted it though.

For you maybe, but not for others. You are aware that the `m` button mutes, right?

**Comment by Qwerty-Space** [comment]
> For you maybe, but not for others. You are aware that the m button mutes, right?

Yes, maybe there should be a `frame-step & hold-mute` thing?

**Comment by Argon-** [comment]
Some commands are executed asynchronously, right? Or more like: they don't necessarily block until their result is in effect.
Otherwise `a set mute true ; frame-step ; set mute false` might work.


(FWIW, I also like it the way it is)

**Comment by Qwerty-Space** [comment]
Thanks, I'll give it a shot

**Comment by Akemi** [comment]
resolved with different key binding.

**Comment by sersorrel** [comment]
fwiw, the suggested solution doesn't seem to work (even when corrected to `set mute yes`/`set mute no`), and nor does applying and restoring a "muted" profile to avoid trashing the existing mute setting.

there's a big set of scripts that claim to include muted stepping: https://github.com/oe-d/control
I haven't tried it, maybe it works, but it would be nice to be able to have just "stepping forwards doesn't play tiny chunks of audio" without having to add nearly 600 lines of lua that does a bunch of other stuff too, and it looks kinda complicated to split out that feature from the rest of that script...

**Comment by haasn** [comment]
Yeah I agree it would be a sane feature to implement. Patches welcome etc.

**Comment by Dudemanguy** [comment]
On master, you can now frame step forward using `frame-step 1 seek` which will do an underlying seek when framestepping so no audio will be played. Of course, it won't be accurate at all for VFR but for most files it should be OK.

**Comment by Dudemanguy** [comment]
There is now a flag to mute frame stepping with 25a0c8307d9302ec4add2f5a8cc07ec2a9934290.
